### ISSUE TITLE
Highlighting for exported functions.

### ISSUE THREAD

**Issue by jack-williams** [ananthakumaran/typescript.el#53]
This isn't a bug so much as a request / query. I'm just wondering about syntax highlighting for functions that are exported.

Given the following example:

```
function foo() {

}

export function bar() {

}
```

`foo` will be coloured but `bar` wont. Was the a conscious choice? Personally I'd like both function names to be coloured (probably using the same colour). I have a hacky solution that changes `typescript--function-heading-1-re` to include an optional `export` string in the regex, however I'm really not knowledgeable enough to know if this is the wrong approach.

(And thanks for the great work on the mode!)

**Comment by josteink** [comment]
Thanks for the bug-report.

I'm going to assume this was an omission. I'll label it a bug, and see if we can't get it fixed.

**Comment by josteink** [comment]
Thanks for the bug-report.

I'm going to assume this was an omission. I'll label it a bug, and see if we can't get it fixed.

If you could show us your changes, maybe we could review them together and see if we find them sufficient. If so, you could always submit them as a PR.

**Comment by jack-williams** [comment]
I've added a new matcher for declare or export modifiers:

```
(defconst typescript--module-declaration-re
  "^\\s-*\\(?:declare\\|\\(?:export\\(?:\\s-+default\\)?\\)\\)?"
  "Regexp matching ambient declaration modifier or export declaration")
```
and then added that to `typescript--function-heading-1-re` as follows:

```
(defconst typescript--function-heading-1-re
  (concat
   typescript--module-declaration-re
   "\\s-*function\\s-+\\(" typescript--name-re "\\)")
  "Regexp matching the start of a typescript function header.
Match group 1 is the name of the function.")
```

**Comment by josteink** [comment]
Sorry about the delay. I've had quite a few other issues I've been trying to close before starting yet another one. :)

Comparing to what is already done in `js-mode`, you have the same exact problem there, so there is no code we can "borrow" for a more proven approach.

`js2-mode` seems structured in a completely different way, and while it has this already solved, I see no obvious way to re-use their solution.

As such, and from what I can tell, that looks perfectly OK. I have no objections to it.

Feel free to send it in as a PR, and other people more versed in font-locking code can help improve it if they see the need.

**Comment by jack-williams** [comment]
Great, I'll put a PR in over the next few days. Thanks!

**Comment by josteink** [comment]
Fixed in #54. Closing.
